iml-manager-cli/src/server.rs

```diff
@@ -373,6 +391,133 @@ pub async fn server_cli(command: ServerCommand) -> Result<(), ImlManagerCliError
 
             wait_for_cmds(cmds).await?;
         }
+        ServerCommand::Remove(config) => {
+            let term = Term::stdout();
+
+            let remove_hosts = hostlist_parser::parse(&config.hosts)?;
+
+            tracing::debug!("Parsed hosts {:?}", remove_hosts);
+
+            let (_, api_hosts) = future::try_join(get_hosts(), get_hosts()).await?;
```

Comment:
You are making two API calls here and throwing one away...

```suggestion
            let api_hosts = get_hosts().await?;
```

Reply:
Done.